src/Disks/IO/AsynchronousReadIndirectBufferFromRemoteFS.cpp

```diff
@@ -231,18 +245,23 @@ off_t AsynchronousReadIndirectBufferFromRemoteFS::seek(off_t offset_, int whence
 
     pos = working_buffer.end();
 
-    /// Note: we read in range [file_offset_of_buffer_end, read_until_position).
-    if (read_until_position && file_offset_of_buffer_end < *read_until_position
-        && static_cast<off_t>(file_offset_of_buffer_end) >= getPosition()
-        && static_cast<off_t>(file_offset_of_buffer_end) < getPosition() + static_cast<off_t>(min_bytes_for_seek))
+    /**
+    * Lazy ignore. Save number of bytes to ignore and ignore it either for prefetch buffer or current buffer.
+    * Note: we read in range [file_offset_of_buffer_end, read_until_position).
+    */
+    off_t file_offset_before_seek = impl->offset();
+    if (impl->initialized()
+        && read_until_position && file_offset_of_buffer_end < *read_until_position
+        && static_cast<off_t>(file_offset_of_buffer_end) > file_offset_before_seek
+        && static_cast<off_t>(file_offset_of_buffer_end) < file_offset_before_seek + static_cast<off_t>(min_bytes_for_seek))
     {
-       /**
-        * Lazy ignore. Save number of bytes to ignore and ignore it either for prefetch buffer or current buffer.
-        */
-        bytes_to_ignore = file_offset_of_buffer_end - getPosition();
+        ProfileEvents::increment(ProfileEvents::RemoteFSLazySeeks);
+        bytes_to_ignore = file_offset_of_buffer_end - file_offset_before_seek;
+        prefetch_future = readInto(prefetch_buffer.data(), prefetch_buffer.size());
```

Comment:
Why we call `readInto` here now? Should we discard the result of previous `prefetch_future`?

Reply:
> Should we discard the result of previous prefetch_future?

It is discarded 20 lines above.

> Why we call readInto here now? 

to ignore redundant now.